test/upgrader_test.dart

```diff
@@ -453,17 +455,96 @@ void main() {
     expect(notCalled, true);
   });
 
-  test('test upgrader shouldDisplayUpgrade', () async {
-    final client = MockClient.setupMockClient();
-    final upgrader = Upgrader();
-    upgrader.client = client;
-    upgrader.debugLogging = true;
+  group('initialize', () {
+    test('should use fake Appcast', () async {
+      final fakeAppcast = FakeAppcast();
+      final client = MockClient.setupMockClient();
+      final upgrader = Upgrader()
+        ..client = client
+        ..appcastConfig = fakeAppcast.config
+        ..debugLogging = true
+        ..appcast = fakeAppcast
+        ..installPackageInfo(
+          packageInfo: PackageInfo(
+            appName: 'Upgrader',
+            packageName: 'com.larryaasen.upgrader',
+            version: '1.9.6',
+            buildNumber: '42',
+          ),
+        );
+
+      await upgrader.initialize();
 
-    expect(upgrader.shouldDisplayUpgrade(), false);
-    upgrader.debugDisplayAlways = true;
-    expect(upgrader.shouldDisplayUpgrade(), true);
-    upgrader.debugDisplayAlways = false;
-    expect(upgrader.shouldDisplayUpgrade(), false);
+      expect(fakeAppcast.callCount, greaterThan(0));
+    });
+  });
+
+  group('shouldDisplayUpgrade', () {
+    test('should respect `debugDisplayAlways` property', () {
+      final client = MockClient.setupMockClient();
+      final upgrader = Upgrader()
+        ..client = client
+        ..debugLogging = true;
+
+      expect(upgrader.shouldDisplayUpgrade(), false);
+      upgrader.debugDisplayAlways = true;
+      expect(upgrader.shouldDisplayUpgrade(), true);
+      upgrader.debugDisplayAlways = false;
+      expect(upgrader.shouldDisplayUpgrade(), false);
+    });
+
+    test('should be blocked when version is below minAppVersion', () async {
+      final upgrader = Upgrader()
+        ..client = MockClient.setupMockClient()
+        ..debugLogging = true
+        ..minAppVersion = '2.0.0'
+        ..installPackageInfo(
+          packageInfo: PackageInfo(
+            appName: 'Upgrader',
+            packageName: 'com.larryaasen.upgrader',
+            version: '1.9.6',
+            buildNumber: '42',
+          ),
+        );
+
+      await upgrader.initialize();
+
+      final shouldDisplayUpgrade = upgrader.shouldDisplayUpgrade();
+
+      expect(shouldDisplayUpgrade, true);
+    });
+
+    test('should be blocked when bestItem has critical update', () async {
```

Comment:
This test does not verify blocked because that is a UI feature. Verifying `showIgnore` and `showLater` are `false` would be the closest thing to validating blocked.

Reply:
updated to `should return true`